include/seqan/blast/blast_statistics.h

```diff
@@ -0,0 +1,1076 @@
+    static double const VALUE[nParamSets][nParams];
+};
+
+template <typename TSpec>
+double const KarlinAltschulValues<Blosum62, TSpec>::VALUE
+  [KarlinAltschulValues<Blosum62, TSpec>::nParamSets]
+  [KarlinAltschulValues<Blosum62, TSpec>::nParams] =
+{
+        {(double) NCBI_INT2_MAX, (double) NCBI_INT2_MAX,
+                (double) NCBI_INT2_MAX, 0.3176, 0.134, 0.4012, 0.7916, -3.2},
+        {11, 2, (double) NCBI_INT2_MAX, 0.297,  0.082, 0.27,   1.1,    -10} ,
+        {10, 2, (double) NCBI_INT2_MAX, 0.291,  0.075, 0.23,   1.3,    -15} ,
+        { 9, 2, (double) NCBI_INT2_MAX, 0.279,  0.058, 0.19,   1.5,    -19} ,
+        { 8, 2, (double) NCBI_INT2_MAX, 0.264,  0.045, 0.15,   1.8,    -26} ,
+        { 7, 2, (double) NCBI_INT2_MAX, 0.239,  0.027, 0.10,   2.5,    -46} ,
+        { 6, 2, (double) NCBI_INT2_MAX, 0.201,  0.012, 0.061,  3.3,    -58} ,
+        {13, 1, (double) NCBI_INT2_MAX, 0.292,  0.071, 0.23,   1.2,    -11} ,
+        {12, 1, (double) NCBI_INT2_MAX, 0.283,  0.059, 0.19,   1.5,    -19} ,
+        {11, 1, (double) NCBI_INT2_MAX, 0.267,  0.041, 0.14,   1.9,    -30} ,
+        {10, 1, (double) NCBI_INT2_MAX, 0.243,  0.024, 0.10,   2.5,    -44} ,
+        { 9, 1, (double) NCBI_INT2_MAX, 0.206,  0.010, 0.052,  4.0,    -87} ,
+};
+/// BLOSUM80
+template <typename TSpec>
+struct KarlinAltschulValues<Blosum80, TSpec>
+{
+    typedef uint8_t size_t;
+
+    /* statics */
+    static constexpr size_t nParams = 8;
+    static constexpr size_t nParamSets = 10;
+    static double const VALUE[nParamSets][nParams];
+};
+
+template <typename TSpec>
+double const KarlinAltschulValues<Blosum80, TSpec>::VALUE
+ [KarlinAltschulValues<Blosum80, TSpec>::nParamSets]
+ [KarlinAltschulValues<Blosum80, TSpec>::nParams] =
+{
+    {(double) NCBI_INT2_MAX, (double) NCBI_INT2_MAX,
+            (double) NCBI_INT2_MAX, 0.3430,  0.177, 0.6568, 0.5222, -1.6},
+    {25, 2, (double) NCBI_INT2_MAX, 0.342,   0.17,  0.66,   0.52,   -1.6},
+    {13, 2, (double) NCBI_INT2_MAX, 0.336,   0.15,  0.57,   0.59,   -3},
+    { 9, 2, (double) NCBI_INT2_MAX, 0.319,   0.11,  0.42,   0.76,   -6},
+    { 8, 2, (double) NCBI_INT2_MAX, 0.308,   0.090, 0.35,   0.89,   -9},
+    { 7, 2, (double) NCBI_INT2_MAX, 0.293,   0.070, 0.27,   1.1,   -14},
+    { 6, 2, (double) NCBI_INT2_MAX, 0.268,   0.045, 0.19,   1.4,   -19},
+    {11, 1, (double) NCBI_INT2_MAX, 0.314,   0.095, 0.35,   0.90,   -9},
+    {10, 1, (double) NCBI_INT2_MAX, 0.299,   0.071, 0.27,   1.1,   -14},
+    { 9, 1, (double) NCBI_INT2_MAX, 0.279,   0.048, 0.20,   1.4,   -19},
+};
+
+/// BLOSUM90
+// not implemented in seqan
+
+/// PAM30
+// not implemented in seqan
+
+/// PAM40
+// not implemented in blast
+
+/// PAM70
+// not implemented in seqan
+
+/// PAM120
+// not implemented in blast
+
+/// PAM200
+// not implemented in blast
+
+/// PAM250
+template <typename TSpec>
+struct KarlinAltschulValues<Pam250, TSpec>
+{
+    typedef uint8_t size_t;
+
+    /* statics */
+    static constexpr size_t nParams = 8;
```

Comment:
Well anything that could be, should be constexpr. If all our types had proper constructors (and the compiler supports c++14), it would be possible to select the KarlinAltschulValues for a given scoring scheme at compile time :)